plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/ProductBlock.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplates;
+
+use Automattic\WooCommerce\Admin\BlockTemplates\BlockInterface;
+use Automattic\WooCommerce\Admin\BlockTemplates\ContainerInterface;
+use Automattic\WooCommerce\Internal\Admin\BlockTemplates\Block;
+
+class ProductBlock extends Block implements ContainerInterface {
+
+	/**
+	 * Adds block to the section block.
+	 *
+	 * @param array $block_config The block data.
+	 */
+	public function &add_block(array $block_config): BlockInterface
+	{
+		$block = new ProductBlock($block_config, $this->get_root_template(), $this);
+		return $this->add_inner_block($block);
+	}
+
+	/**
+	 * Adds block to the section block.
+	 *
+	 * @param array $conditional_attributes Conditional attributes.
+	 * @param array $block_configs The nested block data.
+	 */
+	public function add_conditional_block(array $conditional_attributes, array ...$block_configs): BlockInterface
+	{
+		$conditional_id = '';
+		if ( isset( $block_configs[0]['id'] ) ) {
+			$conditional_id = $block_configs[0]['id'] . '-conditional';
```

Comment:
Edge case, but this will end up with a duplicate ID if the same block is passed to this function twice.

Reply:
I don't think so, I am referencing the `block` id here, if there was a duplicate ID, that would mean the block being nested also has a duplicate ID and would fail anyway.

Reply:
Good point.  Was reading this as the parent block, but i see it's the block being added so duplication, in theory, should not happen.